lib/player.js

```diff
@@ -4211,6 +4215,18 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
             mimeType);
       }
     }
+    if (mimeType === 'text/x-ssa') {
```

Comment:
If you're going to add several formats that do the same things, I would suggest a more generic solution.  For `src=`, I would suggest having us use the text plugins generically to parse the inputs and use common code to convert the text samples to WebVTT.  This is more maintainable than a bunch of if statements and duplicated conversion code.

Reply:
I think you need to talk to @joeyparrish , because when I added LRC I was already setting this up to do what you suggest, but Joey suggested I go the other way. See: https://github.com/google/shaka-player/pull/3036#discussion_r539624474

Reply:
I discussed with @joeyparrish and we agree it would be better to handle this a bit more generically.  I would suggest creating a new `WebVttGenerator` type (preferably with its own tests) to convert Cue objects to WebVTT text.  Then the Player will search for a parser instance to parse the given MIME to create Cue objects.  Then pass that to the WebVttGenerator to convert.

Reply:
Done!